src/System.Management.Automation/engine/Subsystem/FeedbackSubsystem/IFeedbackProvider.cs

```diff
@@ -0,0 +1,285 @@
+    /// Interface for implementing a feedback provider on command failures.
+    /// </summary>
+    public interface IFeedbackProvider : ISubsystem
+    {
+        /// <summary>
+        /// Default implementation. No function is required for a feedback provider.
+        /// </summary>
+        Dictionary<string, string>? ISubsystem.FunctionsToDefine => null;
+
+        /// <summary>
+        /// Default implementation for `ISubsystem.Kind`.
+        /// </summary>
+        SubsystemKind ISubsystem.Kind => SubsystemKind.FeedbackProvider;
+
+        /// <summary>
+        /// Gets feedback based on the given commandline and error record.
+        /// </summary>
+        /// <returns></returns>
+        string? GetFeedback(string commandLine, ErrorRecord lastError, CancellationToken token);
+    }
+
+    internal sealed class GeneralCommandErrorFeedback : IFeedbackProvider
+    {
+        private readonly Guid _guid;
+        private readonly object[] _args;
+        private ScriptBlock? _fuzzySb;
+
+        internal GeneralCommandErrorFeedback()
+        {
+            _guid = new Guid("A3C6B07E-4A89-40C9-8BE6-2A9AAD2786A4");
+            _args = new object[1];
+        }
+
+        public Guid Id => _guid;
+
+        public string Name => "General";
+
+        public string Description => "The built-in general feedback source for command errors.";
+
+        public string? GetFeedback(string commandLine, ErrorRecord lastError, CancellationToken token)
+        {
+            var rsToUse = Runspace.DefaultRunspace;
+            if (rsToUse is null)
+            {
+                return null;
+            }
+
+            if (lastError.FullyQualifiedErrorId == "CommandNotFoundException")
+            {
+                EngineIntrinsics context = rsToUse.ExecutionContext.EngineIntrinsics;
+
+                var target = (string)lastError.TargetObject;
+                CommandInvocationIntrinsics invocation = context.SessionState.InvokeCommand;
+
+                // See if target is actually an executable file in current directory.
+                var localTarget = Path.Combine(".", target);
+                var command = invocation.GetCommand(
+                    localTarget,
+                    CommandTypes.Application | CommandTypes.ExternalScript);
+
+                if (command is not null)
+                {
+                    return StringUtil.Format(
+                        SuggestionStrings.Suggestion_CommandExistsInCurrentDirectory,
+                        target,
+                        localTarget);
+                }
+
+                // Check fuzzy matching command names.
+                if (ExperimentalFeature.IsEnabled("PSCommandNotFoundSuggestion"))
+                {
+                    _fuzzySb ??= ScriptBlock.CreateDelayParsedScriptBlock(@$"
+                        param([string] $target)
+                        $cmdNames = Get-Command $target -UseFuzzyMatch | Select-Object -First 10 -ExpandProperty Name
```

Comment:
Perhaps only show up to 5 with minimal distance of 1

Reply:
Leverage https://github.com/PowerShell/PowerShell/pull/18261

Reply:
It seems `Get-Command -UseFuzzyMatch` already does sorted output. In the case, #18261 looks like an extra.

Reply:
Updated the related scripts to use `-FuzzyMinimumDistance`.